source/common/network/connection_impl.cc

```diff
@@ -39,68 +40,88 @@ void ConnectionImplUtility::updateBufferStats(uint64_t delta, uint64_t new_total
   }
 }
 
-void DynamicSocketOptionsImpl::setSocketRecvBufferSize(uint32_t buffer_size) const {
-  int rc = 0;
-  if (io_handle_.fd() == -1) {
+Api::SysCallIntResult DynamicSocketOptionsImpl::setSocketRecvBufferSize(int buffer_size) const {
+  int rc = -1;
+  int errsrv = -1;
+  socklen_t option_len = sizeof(int);
+  if (-1 == io_handle_.fd()) {
     ENVOY_CONN_LOG(trace, "Stale socket file handle, operation Set Receive Buffer aborted", *this);
```

Comment:
Not so much stale as closed right?

Reply:
I think I borrowed this log from a previous snippet. I might be wrong. I will change this to indicate "Closed socket file handle....". Just was not sure, if -1 always indicated closed.